src/domTree.js

```diff
@@ -21,6 +24,106 @@ const createClass = function(classes: string[]): string {
     return classes.filter(cls => cls).join(" ");
 };
 
+const initNode = function<ChildType>(
+    classes?: string[],
+    children?: ChildType[],
+    options?: Options,
+    style?: CssStyle,
+) {
+    this.classes = classes || [];
+    // $FlowFixMe
+    this.children = children || [];
```

Comment:
You might try doing: `children?: ChildType[] = []` in the param list and then this would become `this.children = children`.

Reply:
@kevinbarabash It still gives following errors:
```
Error --------------------------------------------------------------------------------------------- src/domTree.js:34:21

Cannot assign `children || []` to `this.children` because:
 - `ChildType` [1] is incompatible with `ChildType` [2] in array element.
 - `ChildType` [1] is incompatible with `VirtualNodeInterface` [2] in array element.
 - `ChildType` [1] is incompatible with `HtmlDomNode` [3] in array element.
 - property `classes` is missing in `ChildType` [1] but exists in `HtmlDomNode` [3] in array element.
 - property `depth` is missing in `ChildType` [1] but exists in `HtmlDomNode` [3] in array element.
 - property `hasClass` is missing in `ChildType` [1] but exists in `HtmlDomNode` [3] in array element.
 - property `height` is missing in `ChildType` [1] but exists in `HtmlDomNode` [3] in array element.
 - property `maxFontSize` is missing in `ChildType` [1] but exists in `HtmlDomNode` [3] in array element.
 - property `style` is missing in `ChildType` [1] but exists in `HtmlDomNode` [3] in array element.
 - property `toMarkup` is missing in `ChildType` [1] but exists in `VirtualNodeInterface` [2] in array element.
 - property `toMarkup` is missing in `ChildType` [1] but exists in `VirtualNodeInterface` [3] in array element.
 - property `toNode` is missing in `ChildType` [1] but exists in `VirtualNodeInterface` [2] in array element.
 - property `toNode` is missing in `ChildType` [1] but exists in `VirtualNodeInterface` [3] in array element.
 - property `tryCombine` is missing in `ChildType` [1] but exists in `HtmlDomNode` [3] in array element.
 - `ChildType` [2] is incompatible with `VirtualNodeInterface` [2] in array element.
 - `ChildType` [2] is incompatible with `HtmlDomNode` [3] in array element.

   src/domTree.js:34:21
    34|     this.children = children || [];
                            ^^^^^^^^^^^^^^

References:
   src/domTree.js:29:16
    29|     children?: ChildType[],
                       ^^^^^^^^^ [1]
   src/domTree.js:163:15
   163|     children: ChildType[];
                      ^^^^^^^^^ [2]
   src/domTree.js:217:15
   217|     children: HtmlDomNode[];
                      ^^^^^^^^^^^ [3]
```

Reply:
@marcianx Could you give any insights? Thank you!

Reply:
This earnestly feels broken, and I didn't find a way to cajole the type system to accept it without this override. (I wonder whether typescript would have been a better choice due to greater maturity.) That being said, a simple workaround is to move the `children` assignment into the original constructors:
```
initNode.call(this, classes, options);
this.children = children || [];
```
and remove the type parameter to `initNode` since it's no longer necessary then.

Reply:
@marcianx That seems nice compromise 😄